### PR TITLE
Adding Pktgen test.

### PR DESCRIPTION
**Pull request by sandeepkarthikeyan**
It generates packets for testing NIC and device drives and is
part of linux kernel.

Signed-off-by: Sandeep K <sandeep@linux.vnet.ibm.com>

### CODE REVIEW

File: io/net/pktgen.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        if not os.path.exists('/proc/net/pktgen'):
+            self.error("pktgen not loaded")
+
+        # validating the dst_ip and network interface
+        self.validate_net_interface()
+        self.ping_test()
+
+        # Adding the devices
+        self.log.info("Adding devices")
+        self.pgdev = '/proc/net/pktgen/kpktgend_0'
```
Comment (narasimhan-v) [inline io/net/pktgen.py]: Minor suggestion. I see `/proc/net/pktgen/` being used in multiple places. Can you have it in variable and use..

File: io/net/pktgen.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        shutil.copyfile(self.pgdev, output)
+
+    def pgset(self, command):
+        file_name = open(self.pgdev, 'w')
+        file_name.write(command + '\n')
+        file_name.close()
+        if not self.match_string('Result: OK', self.pgdev):
+            if not self.match_string('Result: NA', self.pgdev):
+                process.system("cat %s" % self.pgdev,
+                               ignore_status=True, shell=True)
```
Comment (narasimhan-v) [inline io/net/pktgen.py]: Isn't this test supposed to fail ?

Comment (PraveenPenguin) [inline io/net/pktgen.py]: @narasimhan-v @sandeepkarthikeyan  i think should be ok as NA means not applicable what you think ?

**Comment by sandeepkarthikeyan** [comment]
Raising a new pull request.

**Pull request closed without merging**